lib/internal/Magento/Framework/File/Uploader.php

```diff
@@ -604,10 +604,12 @@ private function _createDestinationFolder($destinationFolder)
     public static function getNewFileName($destinationFile)
     {
         $fileInfo = pathinfo($destinationFile);
-        if (file_exists($destinationFile)) {
+        $destinationBaseFile = str_replace('/tmp/','/',$destinationFile);
```

Comment:
This function should be abstract and should not rely on the hardcoded path samples like `/tmp/`

Reply:
@sivaschenko please check other comments when doing review

Reply:
@orlangur which comment are you referring to?

Reply:
https://github.com/magento/magento2/pull/20763#discussion_r273345620 answers your question

Reply:
I've seen that comment and the problems mentioned in my review are still actual